### DISCUSSION TITLE
How to modify the body display content of MIMT web without changing the original data of the request

### DISCUSSION THREAD

**Discussion by laojiang2025** [mitmproxy/mitmproxy#5127] [Q&A] [answered]
How to modify the body display content of MIMT web without changing the original data of the request。
My interface, request and respne, are encrypted. I need to see plaintext through packet capture, but I don't want to change the original information of the interface

**Comment by mhils** [comment, marked as answer]
The best way to do this is to implement a custom contentview addon:

- https://github.com/mitmproxy/mitmproxy/blob/main/examples/addons/contentview.py
- https://docs.mitmproxy.org/dev/addons-overview/

**Reply by laojiang2025** [reply]
Thanks , [contentview.py](https://github.com/mitmproxy/mitmproxy/blob/main/examples/addons/contentview.py) Solved my problem, it is very powerful
